src/database/remove.ts

```diff
@@ -0,0 +1,93 @@
+"use strict";
+
+import * as pathLib from "path";
+import { Queue } from "../queue";
+import FirebaseError = require("../error");
+import logger = require("../logger");
+import RemoveRemote from "./remove-remote";
+
+class DatabaseRemove {
+  public path: string;
+  public concurrency: number;
+  public retries: number;
+  public remote: any;
+  private jobQueue: Queue<string>;
+  private waitingPath: Map<string, number>;
+
+  /**
+   * Construct a new RTDB delete operation.
+   *
+   * @constructor
+   * @param {string} path path to delete.
```

Comment:
You can remove `{string}` as the type is documented in the signature. I forget what Typescript does with documentation, but you definitely can keep it if you like (and all the linters let you, I hope)

```suggestion
   * @param path Path to delete.
```